perun/cli.py

```diff
@@ -562,10 +563,17 @@ def log(head: Optional[str], **kwargs: Any) -> None:
     "-sb",
     "format__sort_profiles_by",
     nargs=1,
-    type=click.Choice(profiles.ProfileInfo.valid_attributes),
-    callback=cli_kit.set_config_option_from_flag(pcs.local_config, "format.sort_profiles_by", str),
+    type=str,
+    callback=cli_kit.set_config_option_from_flag(
+        pcs.local_config,
+        "format.sort_profiles_by",
+        functools.partial(
```

Comment:
This is really ugly, and IMO should not be in CLI at all, if you need it like this, then extract it to some function so this partial is hidden.

Reply:
Good point, perhaps the `config.py` module could keep track of allowed values (or other constraints, if any) for individual configuration keys and implement the validity checks instead of cli_kit callbacks?

Reply:
I replaced the `partial` with its alias so that it is more readable. I also created an issue #302 related to the problem of systematic checking config keys and values.

Reply:
Mm, yes, I guess ideal would be to have some json/yaml with the exact specification that would check the validity of the config (together with types, expected ranges, etc.).